### PR TITLE
Update the preset handle field in the SetActivePresetRequest c…

### LINKED ISSUE #35104: Update the SetActivePresetRequestRequest command to have a nullable presetHandle field
The spec change has been made in PR - https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/10144. This issue addresses corresponding changes to SDK.

### PR DESCRIPTION
**Pull request by nivi-apple**
…ommand to be nullable

- Update the deleagte API for getting the active preset handle to return a nullable handle

Fixes: #35104

### CODE REVIEW

**Comment by nivi-apple** [comment]
Added skip-protocol-compatibility since we are updating command payload for SetActivePresetRequest to be nullable. spec change merged - https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/10144

**Review by hasty — APPROVED** [review]

File: src/app/clusters/thermostat-server/thermostat-delegate.h [hunk 1]
```diff
@@ -78,11 +78,10 @@ class Delegate
...
      * @brief Get the ActivePresetHandle attribute value.
      *
-     * @param[out] activePresetHandle The MutableByteSpan to copy the active preset handle into. On success,
-     *             the callee must update the length to the length of the copied data. If the value of
-     *             the attribute is null, the callee must set the MutableByteSpan to empty.
+     * @param[out] activePresetHandle The nullable MutableByteSpan to copy the active preset handle into. On success,
+     *             the size of the activePresetHandle is updated to the length of the copied data.
      */
-    virtual CHIP_ERROR GetActivePresetHandle(MutableByteSpan & activePresetHandle) = 0;
+    virtual CHIP_ERROR GetActivePresetHandle(DataModel::Nullable<MutableByteSpan> & activePresetHandle) = 0;
```
Comment (bzbarsky-apple) [inline src/app/clusters/thermostat-server/thermostat-delegate.h:84]: For what it's worth, keeping this as "zero length means null" would have been OK too... but fine.

File: src/app/clusters/thermostat-server/thermostat-server.cpp [hunk 2]
```diff
@@ -761,19 +761,13 @@ CHIP_ERROR ThermostatAttrAccess::Read(const ConcreteReadAttributePath & aPath, A
         VerifyOrReturnError(delegate != nullptr, CHIP_ERROR_INCORRECT_STATE, ChipLogError(Zcl, "Delegate is null"));
 
         uint8_t buffer[kPresetHandleSize];
-        MutableByteSpan activePresetHandle(buffer);
+        MutableByteSpan activePresetHandleSpan(buffer);
+        DataModel::Nullable<MutableByteSpan> activePresetHandle = DataModel::MakeNullable(activePresetHandleSpan);
```
Comment (bzbarsky-apple) [inline src/app/clusters/thermostat-server/thermostat-server.cpp:765]:
```suggestion
        auto activePresetHandle = DataModel::MakeNullable(activePresetHandleSpan);
```

File: src/app/clusters/thermostat-server/thermostat-server.cpp [hunk 3]
```diff
@@ -1470,7 +1464,8 @@ imcode commitPresets(Delegate * delegate, EndpointId endpoint)
     // attribute. If a preset is not found with the same presetHandle, return INVALID_IN_STATE. If there is no ActivePresetHandle
     // attribute set, continue with other checks.
     uint8_t buffer[kPresetHandleSize];
-    MutableByteSpan activePresetHandle(buffer);
+    MutableByteSpan activePresetHandleSpan(buffer);
+    DataModel::Nullable<MutableByteSpan> activePresetHandle = DataModel::MakeNullable(activePresetHandleSpan);
```
Comment (bzbarsky-apple) [inline src/app/clusters/thermostat-server/thermostat-server.cpp:1468]:
```suggestion
    auto activePresetHandle = DataModel::MakeNullable(activePresetHandleSpan);
```

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/clusters/thermostat-server/thermostat-delegate.h (hunk 1 above)
Comment (nivi-apple) [reply src/app/clusters/thermostat-server/thermostat-delegate.h:84]: yeah i thought of that but wanted to just make it explicit.

**Review by ReneJosefsen — APPROVED** [review]

**Pull request merged**